checkov/terraform/graph_builder/variable_rendering/evaluate_terraform.py

```diff
@@ -167,6 +172,85 @@ def evaluate_compare(input_str: str) -> Union[str, bool]:
     return input_str
 
 
+def handle_for_loop_in_foreach(input_str: str) -> str:
+    # not implementing for loop with condition for now
+    if isinstance(input_str, str) and renderer.FOR_LOOP in input_str and '?' not in input_str:
+        old_input_str = input_str
+        try:
+            e = ast.literal_eval(input_str)
+            if isinstance(e, list) and len(e) == 1:
+                input_str = e[0]
+        except (ValueError, SyntaxError):
+            pass
+        if isinstance(input_str, str) and renderer.FOR_LOOP in input_str:
```

Comment:
again about isinstance.
Also, I think you can seperate each block of code under the `if` statement to a function with indicative name.
For example, the above block will be `_handle_literal` and this block will be `handle_<whatever-it-handles>`

Reply:
For isinstace - we override the input_str so, maybe is not a string.